cargo-pgrx/src/command/upgrade.rs

```diff
@@ -0,0 +1,216 @@
+//LICENSE Portions Copyright 2019-2021 ZomboDB, LLC.
+//LICENSE
+//LICENSE Portions Copyright 2021-2023 Technology Concepts & Design, Inc.
+//LICENSE
+//LICENSE Portions Copyright 2023-2023 PgCentral Foundation, Inc. <contact@pgcentral.org>
+//LICENSE
+//LICENSE All rights reserved.
+//LICENSE
+//LICENSE Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+use cargo_edit::{registry_url, Dependency};
+use eyre::eyre;
+use std::path::{Path, PathBuf};
+use toml_edit::KeyMut;
+use tracing::{debug, error, info, warn};
+
+use crate::CommandExecute;
+
+/// Upgrade pgrx crate versions in `Cargo.toml`.
+/// Defaults to latest.
+#[derive(clap::Args, Debug)]
+#[clap(author)]
+pub(crate) struct Upgrade {
+    /// Specify a version to upgrade to, rather than defaulting to the latest
+    /// version.
+    #[clap(long, short)]
+    pub(crate) target_version: Option<String>, //TODO: typed version not string
+
+    /// Path to the manifest file (usually Cargo.toml). Defaults to
+    /// "./Cargo.toml" in the working directory.
+    #[clap(long, short)]
+    pub(crate) path: Option<PathBuf>,
+
+    // Flag to permit pre-release builds
+    #[clap(long, short)]
+    pub(crate) allow_prerelease: bool,
```

Comment:
hmm. so, cargo-upgrade doesn't have a flag that corresponds to this - it will only search stable versions unless you explicitly say the version of the pre-release you want to upgrade to. i kinda like that behavior, i think people should either know which pre-release they're going to use or just use stable. can you talk more about the use case for "the latest pre-release"?

Reply:
Well, the first use-case I can think of is dogfooding - I can imagine while testing our own projects it could be handy. However, I can also understand why most end-users would be unlikely to use this feature. Is there any downside to offering it as a feature?

Reply:
eh? while reading this conversation, I wondered if I slightly misinterpreted the name of this. it first seemed that this flag was required for using a prerelease at all? is it only for setting the latest prerelease?

maybe should be `--include-prereleases` instead? or just `--prerelease`

Reply:
Oh that's less ambiguous naming, I agree.

Reply:
This should be dealt with in the latest revision. I've renamed it to "include-prereleases" and disabled the short-form argument (which would be "-i")